src/pydocstyle/parser.py

```diff
@@ -373,9 +380,12 @@ def parse_definitions(self, class_, all=False):
         while self.current is not None:
             self.log.debug("parsing definition list, current token is %r (%s)",
                            self.current.kind, self.current.value)
+            self.log.debug('got_newline: %s', self.stream.got_newline)
             if all and self.current.value == '__all__':
                 self.parse_all()
-            elif self.current.kind == tk.OP and self.current.value == '@':
+            elif (self.current.kind == tk.OP and
+                  self.current.value == '@' and
+                  self.stream.got_newline):
```

Comment:
I'm having trouble understanding why this works. As far as I understand, `stream.got_newline` only specifies if the previous token was a newline, not if the whole line was just an empty new line. Let's take your test case:
```python
(a
@b)
@foo
def bar():
    ...
```

If what I said is correct, how does this condition fails to detect `@b)` as another decorator? I'm guessing that it's something I didn't get about the way the parser works :^)